test/adjtrans.jl

```diff
@@ -809,4 +809,23 @@ end
     end
 end
 
+@testset "lmul!/rmul! by numbers" begin
```

Comment:
Do we have some sort of `Quaternions` available? We should test this on something that is not commutative, because otherwise the whole left- and right-multiplication difference doesn't become visible.

Reply:
Turns out that this only holds for commutative numbers, so I have restricted this to real and complex numbers. This should cover most use cases.

Reply:
For the adjoint case, this should work, because we still have `(xy)' = y'x'`.